plugins/assemblyai/src/stt.ts

```diff
@@ -0,0 +1,195 @@
+import { type AudioBuffer, AudioByteStream, AudioEnergyFilter, log, stt } from '@livekit/agents';
+import type { AudioFrame } from '@livekit/rtc-node';
+import { AssemblyAI } from 'assemblyai';
+import type { RealtimeTranscriber } from 'assemblyai';
+
+export interface STTOptions {
+	apiKey?: string;
+	interimResults: boolean;
+	sampleRate: number;
+	keywords: [string, number][];
+	endUtteranceSilenceThreshold?: number;
+}
+
+const defaultSTTOptions: STTOptions = {
+	apiKey: process.env.ASSEMBLY_AI_KEY,
+	interimResults: true,
+	sampleRate: 16000,
+	keywords: [],
+	// NOTE:
+	// The default is 700ms from AssemblyAI.
+	// I use a low default of 300ms here because I also use
+	// the new end-of-utterance model from LiveKit to handle
+	// turn detection in my agent. Which means that even though
+	// this will quickly return a final transcript EVEN THOUGH
+	// USER IS NOT DONE SPEAKING, the EOU model from LiveKit
+	// DOES properly differentiate and doesn't interrupt (magically!)
+	// Ref: https://blog.livekit.io/using-a-transformer-to-improve-end-of-turn-detection/
+	endUtteranceSilenceThreshold: 200,
+};
+
+export class STT extends stt.STT {
+	#opts: STTOptions;
+	#logger = log();
+	label = 'assemblyai.STT';
+
+	constructor(opts: Partial<STTOptions> = defaultSTTOptions) {
+		super({
+			streaming: true,
+			interimResults: opts.interimResults ?? defaultSTTOptions.interimResults,
+		});
+		if (opts.apiKey === undefined && defaultSTTOptions.apiKey === undefined) {
+			throw new Error(
+				'AssemblyAI API key is required, whether as an argument or as $ASSEMBLY_AI_KEY',
+			);
+		}
+
+		this.#opts = { ...defaultSTTOptions, ...opts };
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	async _recognize(_: AudioBuffer): Promise<stt.SpeechEvent> {
+		throw new Error('Recognize is not supported on AssemblyAI STT');
+	}
+
+	stream(): stt.SpeechStream {
+		return new SpeechStream(this, this.#opts);
+	}
+}
+
+export class SpeechStream extends stt.SpeechStream {
+	#opts: STTOptions;
+	#audioEnergyFilter: AudioEnergyFilter;
+	#logger = log();
+	#speaking = false;
+	#client: AssemblyAI;
+	#transcriber?: RealtimeTranscriber;
+	label = 'assemblyai.SpeechStream';
+
+	constructor(stt: STT, opts: STTOptions) {
+		super(stt);
+		this.#opts = opts;
+		this.closed = false;
+		this.#audioEnergyFilter = new AudioEnergyFilter();
+		this.#client = new AssemblyAI({
+			apiKey: this.#opts.apiKey || '',
+		});
+
+		this.#run();
+	}
+
+	async #run() {
+		try {
+			// Create the realtime transcriber with parameters that AssemblyAI supports
+			this.#transcriber = this.#client.realtime.transcriber({
+				sampleRate: this.#opts.sampleRate,
+				wordBoost: this.#opts.keywords.map((k) => k[0]),
+				endUtteranceSilenceThreshold: this.#opts.endUtteranceSilenceThreshold,
+			});
+
+			// Set up event handlers
+			this.#transcriber.on('open', (data) => {
+				this.#logger.debug(
+					`AssemblyAI session opened: ${data.sessionId}, expires at: ${data.expiresAt}`,
```

Comment:
you should use `logger.child({ ... }).debug(...)` to print the `sessionId` and `expiresAt` instead, for better parsing if someone needs it from the log